src/main/java/com/example/school/SchoolApplication.java

```diff
@@ -492,11 +493,22 @@ private static List<Activity> findActivitiesForSpecificGroup(List<Activity> acti
     }
 
     //method that get a list of students who was absent in specific day
-    private static Set<UUID> findStudentsAbsentInSpecificDay(List<Activity> activities, OffsetDateTime date) {
+    private static Set<UUID> findStudentsAbsentAtSpecificDate(List<Activity> activities, OffsetDateTime date) {
         return activities.stream()
                 .filter(e -> e.getLesson().getStartDate().toLocalDate().equals(date.toLocalDate()))
                 .filter(e -> !e.isPresent())
                 .map(e -> e.getStudent().getId())
                 .collect(Collectors.toSet());
     }
+
+    //method that get the number of lessons taught by a particular teacher on a particular day
+    private static long findNumberLessonsFromTeacherAtSpecificDay
+                                                (List<Activity> activities, UUID teacherId, OffsetDateTime date) {
+        Set<Lesson> lessonsFromTeacherAtThisDate = activities.stream()
+                .map(Activity::getLesson)
+                .filter(lesson -> lesson.getStartDate().toLocalDate().equals(date.toLocalDate()))
+                .filter(lesson -> lesson.getTeacher().getId().equals(teacherId))
+                .collect(Collectors.toSet());
+        return lessonsFromTeacherAtThisDate.size();
```

Comment:
You can use one stream without method `size()`. Try to use method `count()` in Stream API.